community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/CostCompatibility.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compatibility.v3_3
+
+import java.time.Clock
+
+import org.neo4j.cypher.internal.compatibility.v3_4.{MonitoringCacheAccessor, WrappedMonitors => WrappedMonitorsV3_4}
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime._
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime.executionplan.{ExecutionPlan => ExecutionPlanV3_4}
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime.phases.CompilationState
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime.{CommunityRuntimeContext => CommunityRuntimeContextV3_4}
+import org.neo4j.cypher.internal.compiler.v3_3
+import org.neo4j.cypher.internal.compiler.v3_3.{CypherCompilerFactory, DPPlannerName => DPPlannerNameV3_3, IDPPlannerName => IDPPlannerNameV3_3}
+import org.neo4j.cypher.internal.compiler.v3_4._
+import org.neo4j.cypher.internal.compiler.v3_4.phases.LogicalPlanState
+import org.neo4j.cypher.internal.frontend.v3_3.ast.{Statement => StatementV3_3}
+import org.neo4j.cypher.internal.frontend.v3_3.phases.{Monitors => MonitorsV3_3}
+import org.neo4j.cypher.internal.frontend.v3_4.phases.{Monitors, Transformer}
+import org.neo4j.cypher.{CypherPlanner, CypherRuntime, CypherUpdateStrategy}
+import org.neo4j.kernel.monitoring.{Monitors => KernelMonitors}
+import org.neo4j.logging.Log
+
+case class CostCompatibility[CONTEXT3_3 <: v3_3.phases.CompilerContext,
```

Comment:
This is the only implementation of the `Compatibility[CONTEXT3_3, CONTEXT3_4, T]` trait. Why do we have this division? The trait is not only declaring an interface, it contains a whole lot of code. I believe this made sense back in the days where we had both CostCompatablity and RuleCompatability, but I don't think we should perpetuate that pattern any longer.